graylog2-web-interface/src/components/common/EntityDataTable/contexts/SelectedEntitiesProvider.tsx

```diff
@@ -25,18 +25,22 @@ import type { EntityBase } from '../types';
 
 type Props<Entity extends EntityBase> = React.PropsWithChildren<{
   initialSelection?: Array<string>,
-  onChangeSelection: (selectedEntities: Array<Entity['id']>) => void
+  onChangeSelection: (selectedEntities: Array<Entity['id']>, data: Readonly<Array<Entity>>) => void,
+  entities: Readonly<Array<Entity>>,
 }>
 
-const SelectedEntitiesProvider = <Entity extends EntityBase>({ children, initialSelection, onChangeSelection }: Props<Entity>) => {
+const SelectedEntitiesProvider = <Entity extends EntityBase>({ children, initialSelection, onChangeSelection, entities }: Props<Entity>) => {
   const [selectedEntities, setSelectedEntities] = useState<Array<Entity['id']>>(initialSelection);
```

Comment:
Just an idea. we can add also toggleSelectedEntity method and use it in TableRow instead of toggleRowSelect